File: dataset/human/java/file_227.java
import java.util.Scanner;

import java.lang.Math;

public class Main {
  private static final int MAX_A = 1000000000;

  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    int N = scanner.nextInt();

    int num_neg = 0;
    long sum_abs = 0;
    int min_abs = MAX_A;
    for (int i = 0; i < N; i++) {
      int A = scanner.nextInt();
      num_neg += A < 0 ? 1 : 0;
      sum_abs += Math.abs(A);
      min_abs = Math.min(min_abs, Math.abs(A));
    }

    long ans;
    if (num_neg % 2 == 0)
      ans = sum_abs;
    else
      ans = sum_abs - min_abs * 2;
    System.out.println(ans);
  }
}

File: dataset/human/c/file_227.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    int n,a[100000],minus_count = 0;
    int i,j;

    scanf("%d",&n);
    for (i = 0; i < n; i++)
    {
        scanf("%d",&a[i]);
        if(a[i] < 0)minus_count++;
    }
    
    long long ans = 0,min = 2000000000;
    for (i = 0; i < n; i++)
    {
        ans += abs(a[i]);
        if(min > abs(a[i]))min = abs(a[i]);
    }
    
    if(minus_count % 2 == 0)
    {
        printf("%lld",ans);
    }
    else
    {
        printf("%lld",ans - min * 2);
    }
    
}
